pkg/gameserversets/gameserversets.go

```diff
@@ -120,14 +120,35 @@ func sortGameServersByLeastFullNodes(list []*agonesv1.GameServer, count map[stri
 	return list
 }
 
-// sortGameServersByNewFirst sorts by newest gameservers first, and returns them
-func sortGameServersByNewFirst(list []*agonesv1.GameServer) []*agonesv1.GameServer {
+// sortGameServersByNewFirst sorts by newest gameservers first.
+// If FeatureCountsAndLists is enabled, sort by Priority first, then tie break with newest gameservers.
+func sortGameServersByNewFirst(list []*agonesv1.GameServer, priorities []agonesv1.Priority) []*agonesv1.GameServer {
 	sort.Slice(list, func(i, j int) bool {
 		a := list[i]
 		b := list[j]
 
-		// TODO: Sort by Priority First, then tie break with NewFirst
-
+		if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+			for _, priority := range priorities {
+				res := compareGameServerCapacity(&priority, a, b)
+				switch priority.Order {
+				case agonesv1.GameServerPriorityAscending:
+					if res == -1 {
+						return true
+					}
+					if res == 1 {
+						return false
+					}
+				case agonesv1.GameServerPriorityDescending:
+					if res == -1 {
+						return false
+					}
+					if res == 1 {
+						return true
+					}
+				}
+			}
+		}
+		// TODO: Do we still want to keep this as the tie-breaker?
```

Comment:
Do we still want to keep time as the tie-breaker for sorting by Priorities?

Reply:
Yes I think we do. Always good to have a final tie breaker. Also - they may not have priorities!